server/db/driver/pg/orders.go

```diff
@@ -240,14 +240,14 @@ func (a *Archiver) FlushBook(base, quote uint32) (sellsRemoved, buysRemoved []or
 		fail()
 		return
 	}
+	defer rows.Close()
 
```

Comment:
This one is a false positive, but it's better to do this defer like you suggest.  From https://golang.org/pkg/database/sql/#Rows.Close:

> If Next is called and returns false and there are no further result sets, the Rows are closed automatically and it will suffice to check the result of Err.

That's what's happening below.  If `Next` returns false, `Rows` are automatically closed, we call `rows.Err`.  If `Scan` errors, we call `Close` before returning.  However, `Close` is idempotent and we've clearly missed calling it like this in several other places so I'm good with a defer even if it is redundant when `Next` returns false.